Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/FeedbackForm.hs

```diff
@@ -0,0 +1,89 @@
+{-
+ Copyright 2022-23, Juspay India Pvt Ltd
+
+ This program is free software: you can redistribute it and/or modify it under the terms of the GNU Affero General Public License
+
+ as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version. This program
+
+ is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+
+ or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero General Public License for more details. You should have received a copy of
+
+ the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
+-}
+
+module Domain.Action.UI.FeedbackForm where
+
+import Data.Foldable ()
+import qualified Domain.Types.Feedback as DFeedback
+import Domain.Types.FeedbackForm (FeedbackAnswer (FeedbackAnswer), FeedbackFormReq)
+import qualified Domain.Types.Person as DP
+import qualified Domain.Types.Ride as DRide
+import Environment
+import qualified EulerHS.Language as L
+import EulerHS.Prelude hiding (id)
+import qualified Kernel.Storage.Esqueleto as Esq
+import Kernel.Types.APISuccess (APISuccess (Success))
+import Kernel.Types.Error
+import Kernel.Types.Id
+import Kernel.Utils.Common hiding (id)
+import qualified Storage.Queries.Feedback as QFeedback
+import qualified Storage.Queries.FeedbackBadge as QFeedbackBadge
+import qualified Storage.Queries.Ride as QRide
+
+saveFeedbackFormResult :: FeedbackFormReq -> Flow APISuccess
+saveFeedbackFormResult feedbackFormReq = do
+  let rideId = feedbackFormReq.orderId
+  ride <- Esq.runInReplica $ QRide.findById rideId >>= fromMaybeM (RideDoesNotExist rideId.getId)
+  feedbackChipsList <- getFeedbackAnswers feedbackFormReq
+  addFeedbackToTable feedbackChipsList rideId ride.driverId
+  updateFeedbackBadgeToTable feedbackChipsList ride.driverId
```

Comment:
Multiple case match are there and one function is just adding the values to db while other is checking the values from different table and updating the count if data is present otherwise adding the new data.